internal/credentials/gcp/secretmanager.go

```diff
@@ -0,0 +1,158 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"google.golang.org/api/option"
+
+	"github.com/chainloop-dev/chainloop/internal/credentials"
+	"github.com/chainloop-dev/chainloop/internal/servicelogger"
+	"github.com/docker/distribution/uuid"
+	"github.com/go-kratos/kratos/v2/log"
+	"github.com/googleapis/gax-go/v2"
+)
+
+type SecretsManagerInterface interface {
+	CreateSecret(ctx context.Context, req *secretmanagerpb.CreateSecretRequest, opts ...gax.CallOption) (*secretmanagerpb.Secret, error)
+	AddSecretVersion(ctx context.Context, req *secretmanagerpb.AddSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.SecretVersion, error)
+	AccessSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error)
+	DeleteSecret(ctx context.Context, req *secretmanagerpb.DeleteSecretRequest, opts ...gax.CallOption) error
+	GetSecret(ctx context.Context, req *secretmanagerpb.GetSecretRequest, opts ...gax.CallOption) (*secretmanagerpb.Secret, error)
+	Close() error
+}
+
+type Manager struct {
+	projectID string
+	client    SecretsManagerInterface
+	logger    *log.Helper
+}
+
+type NewManagerOpts struct {
+	ProjectID, AuthKey string
+	Logger             log.Logger
+}
+
+func NewManager(opts *NewManagerOpts) (*Manager, error) {
+	if opts.ProjectID == "" || opts.AuthKey == "" {
+		return nil, errors.New("projectID and authKey are required")
+	}
+
+	l := opts.Logger
+	if l == nil {
+		l = log.NewStdLogger(io.Discard)
+	}
+
+	logger := servicelogger.ScopedHelper(l, "credentials/gcp-secrets-manager")
```

Comment:
Thanks for initializing this logger but it seems to be underutilized. I know that the AWS backend doesn't use it much either (which I'll change it soon), but the vault backend does. It logs when a secret is created, read and deleted. You might want to add these log entries too either here or in another patch.

i.e https://github.com/chainloop-dev/chainloop/blob/a812ba2aadf8e1075f51c34786f8d241a9d47a87/internal/credentials/vault/keyval.go#L113

Reply:
Very good point added logging for common operations.